### ISSUE TITLE
Solid vertical lines to reach 30fps

### ISSUE THREAD

**Issue by maxxoccupancy** [jhhoward/MegaDOOM#5]
Instead of storing prescaled textures, how about just two versions of each vertical line.  We generate two lists:

1. DMA list for upcoming long writes (lines more than 10-15 pixels long)
2. CPU writes that are only a few pixels high, so aren't worth the DMA overhead

The VDP just draws what's in its 20KB alternating frame buffers (foreground) over the background layer (just the sky/mountains above the mid point and the floor beneath it).  Instead of 64x64 textures, we have two lists of 64 colors 

Alternately, I've noticed that a number of games like Starfox DMA, StarCruiser, Hard Drivin, etc, draw quads or triangles, and then fill them.  IIRC, the Genie/MD VDP has some method of very quickly filling in shapes that doesn't eat up too much performance.  So some large areas could just get a one-color fill, such as distant walls and stairs that just need to be drawn.

The _"quadrilateral curtains"_ at least take the same shape as DOOM.  And when Doom Guy stops moving, we have a few frames to quickly draw an accurate, scaled, lit, rendered view of the scene.  We might even use color cycling to get the ceiling and floors when we simply move forward, as players do about 60-70% of the time.

20KB represents a block of 200x200 or 256x160.  Using part of the ceiling and floors in both buffers, we could get to 320x160 or maybe even 320x176, streaming in bad guy sprites from the cart.

Bottom line, this would not look like the wonderful DOOM textures, but a bit like a more advanced version of Wolfenstein with variable height walls, stairs, etc.  At least using Shadow/Highlight Mode to create variable shading and lights, we' get something that plays like DOOM.  People should understand that we're more than a full generation below the minimum system requirements here.

**Comment by maxxoccupancy** [comment]
![image](https://github.com/user-attachments/assets/3e4b084c-e10d-4c94-91f5-2904d70afd4b)

So we can't get all of these weird, cool shapes while moving.  We can only get very basic (almost solid color) vertical line versions with the 68000 calculating all of the start and stop points for the DMA, then drawing all of the short lines that will get transferred over during HBlank.

We don't have a 486 20MHz, but we can use the 68000 to start work on nextFrame, the DMA like a Load Store Unit to start drawing thisFrame (the frame being worked on now, but not yet finished for display), and the VDP like an Integer Unit to draw currentFrame.
